source/braille.py

```diff
@@ -27,73 +27,181 @@
-	controlTypes.STATE_EXPANDED: _("-"),
-	# Translators: Displayed in braille when an object (e.g. a tree view item) is collapsed.
-	controlTypes.STATE_COLLAPSED: _("+"),
+	# Displayed in braille when an object (e.g. a toggle button) is pressed.
+	controlTypes.STATE_PRESSED: u"⢎⣿⡱",
+	# Displayed in braille when an object (e.g. a check box) is checked.
+	controlTypes.STATE_CHECKED: u"⢎⣿⡱",
+	# Displayed in braille when an object (e.g. a check box) is half checked.
+	controlTypes.STATE_HALFCHECKED: u"⢎⣭⡱",
 	# Translators: Displayed in braille when an object (e.g. an editable text field) is read-only.
 	controlTypes.STATE_READONLY: _("ro"),
+	# Displayed in braille when an object (e.g. a tree view item) is expanded.
+	controlTypes.STATE_EXPANDED: "-",
+	# Displayed in braille when an object (e.g. a tree view item) is collapsed.
+	controlTypes.STATE_COLLAPSED: "+",
+	# Translators: Displayed in braille when an object has a popup (usually a sub-menu).
+	controlTypes.STATE_HASPOPUP: _("submnu"),
+	# Displayed in braille when a protected control or a document is encountered.	
+	controlTypes.STATE_PROTECTED: "***",
+	# Translators: Displayed in braille when a required form field is encountered.
+	controlTypes.STATE_REQUIRED: _("req"),
+	# Translators: Displayed in braille when an invalid entry has been made.
+	controlTypes.STATE_INVALID_ENTRY: _("invalid"),
+	# Displayed in braille when an object supports autocompletion.
+	controlTypes.STATE_AUTOCOMPLETE: "...",
+	# Translators: Displayed in braille when an edit field allows typing multiple lines of text such as comment fields on websites.
+	controlTypes.STATE_MULTILINE: _("mln"),
 	# Translators: Displayed in braille when an object is clickable.
 	controlTypes.STATE_CLICKABLE: _("clk"),
+	# Translators: Displayed in braille when an object is sorted ascending.
+	controlTypes.STATE_SORTED_ASCENDING: _("sorted asc"),
+	# Translators: Displayed in braille when an object is sorted descending.
+	controlTypes.STATE_SORTED_DESCENDING: _("sorted desc"),
+	# Translators: Displayed in braille when an object (usually a graphic) has a long description.
+	controlTypes.STATE_HASLONGDESC: _("ldesc"),
+	# Translators: Displayed in braille when there is a formula on a spreadsheet cell.
+	controlTypes.STATE_HASFORMULA: _("frml"),
+	# Translators: Displayed in braille when there is a comment for a spreadsheet cell or piece of text in a document.
+	controlTypes.STATE_HASCOMMENT: _("cmnt"),
 }
 negativeStateLabels = {
-	# Translators: Displayed in braille when an object (e.g. a check box) is not checked.
-	controlTypes.STATE_CHECKED: _("( )"),
+	# Translators: Displayed in braille when an object is not selected.
+	controlTypes.STATE_SELECTED: _("nsel"),
+	# Displayed in braille when an object (e.g. a toggle button) is not pressed.
+	controlTypes.STATE_PRESSED: u"⢎⣉⡱",
+	# Displayed in braille when an object (e.g. a check box) is not checked.
+	controlTypes.STATE_CHECKED: u"⢎⣉⡱",

```

Comment:
To me ⣉ looks distinctly taller than x.
SuperNova uses ⣏⣀⣹ to represent an empty checkbox. Maybe that makes it clearer that it is an empty box rather than an x? In the case of NVDA we could stick to the more 'rounded' box ⢎⣀⡱ which I think is a bit easier on the fingers.